THREADING.md

```diff
@@ -78,7 +78,10 @@ platform-native mutex mechanisms to perform the locking, which you may
 rely on if you do not want to use OpenSSL outside of libgit2, or you
 know that libgit2 will outlive the rest of the operations. It is not
 safe to use OpenSSL multi-threaded after libgit2's shutdown function
-has been called.
+has been called.  Note `git_openssl_set_locking()` only works if
+libgit2 uses OpenSSL directly (if GIT_OPENSSL is defined)
+- if OpenSSL is only used as a dependency of libssh2 as described below,
+`git_openssl_set_locking()` is a no-op.
```

Comment:
We mention libgit2 not linking against OpenSSL later on. We should either put that there, or merge the paragraphs.